tests/test_datetimes.py

```diff
@@ -759,14 +759,25 @@ def test_time_ns():
     utc_time = local_time - datetime.timedelta(seconds=time.timezone)
     expected_timestamp = time.mktime(utc_time.timetuple())
 
-    freezer.start()
-    assert time.time() == expected_timestamp
-    assert time.time_ns() == expected_timestamp * 1e9
-    freezer.stop()
+    with freezer:
```

Comment:
This is totally unrelated: this test will crash pytest if the time_ns() function raises an error

Reply:
I'm not sure I understand your comment. On an error, wouldn't the `__exit__` code run, and thus NOT cause any other test side effects? I'm failing to see an issue, could you please elaborate?

Reply:
The issue was present _before_ the change. My comment should have said "did" crash and not "will" crash 😑